flow/activities/flowable_core.go

```diff
@@ -202,24 +200,25 @@ func syncCore[TPull connectors.CDCPullConnectorCore, TSync connectors.CDCSyncCon
 				a.Alerter.LogFlowError(ctx, flowName, err)
 			}
 			if temporal.IsApplicationError(err) {
-				return 0, err
+				return nil, err
 			} else {
-				return 0, fmt.Errorf("failed in pull records when: %w", err)
+				return nil, fmt.Errorf("failed in pull records when: %w", err)
 			}
 		}
 		logger.Info("no records to push")
+		syncWaiting.Store(true)
```

Comment:
why would this be true? also unsure how we hit case in 208

Reply:
`Waiting` should generally be true as long as a sync/normalize isn't happening. Should probably initialize them as true. For normalize I settled on having that be true while channel is being waited on, with syncWaiting it's less clear, maybe a string saying what state it's in would be better, between setup, sync, schema deltas, & waiting on normalize

Reply:
sync being a string makes sense; or you could add an `Idle` state